pkg/controller/enrollment/enroller.go

```diff
@@ -53,17 +53,20 @@ type enroller struct {
 }
 
 func newEnroller(scope scope.Scope, leader func() stack.Leadership, options enrollment.Options) (*enroller, error) {
-
 	l := &enroller{
 		leader:  leader,
 		scope:   scope,
 		options: options,
 	}
-
-	if l.options.SyncInterval.Duration() == 0 {
-		return nil, fmt.Errorf("sync interval cannot be 0")
+	if err := validateParseErrorOptions(l.options); err != nil {
+		return nil, err
+	}
+	// Note that the sync interval is valided in the constructor only since it cannot
+	// be modified after the enroller has started (ie, it cannot be changed on a spec
+	// update).
+	if l.options.SyncInterval.Duration() <= 0 {
```

Comment:
How about we consolidate this duration check with the parse error options validation into one function and have all the logic there?

Reply:
@chungers see comment right above:
```
	// Note that the sync interval is valided in the constructor only since it cannot
	// be modified after the enroller has started (ie, it cannot be changed on a spec
	// update).
```
That's why I kept it out; thoughts?

Reply:
Just a thought...

How about we do something like this in `enrollment/types/Options`

```
type PluginPhase int

const (
	PluginInit = iota
	PluginCommit
)

func (o Options) Validate(phase PluginPhase) error {
   if phase == PluginInit {
      // check the sync interval here too
   }
}
```
This validate function will tell me what's going on in terms of what is being checked during what lifecycle phases of the plugin.  We won't have validation code peppered throughout the enroller and having to figure out what's going on by understanding the call flow...
  
Also, with this we are adding some formalism to options validation and we might want to push this up to make `Validate(phase)` an interface that all the Options can implement.  Then we can move some of this code later out of the enroller.go and have the basic controller implementation enforce this validation...   

Thoughts?

Reply:
I like this idea; having a more formalized `Option` validation scheme sounds like a great idea.